src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/UpdateCartHandler.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\CommandHandler\Checkout;
+
+use Sylius\Bundle\ApiBundle\Assigner\OrderPromoCodeAssignerInterface;
+use Sylius\Bundle\ApiBundle\Command\Checkout\UpdateCart;
+use Sylius\Bundle\ApiBundle\Modifier\OrderAddressModifierInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Symfony\Component\Messenger\Handler\MessageHandlerInterface;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+class UpdateCartHandler implements MessageHandlerInterface
+{
+    private OrderRepositoryInterface $orderRepository;
+
+    private OrderAddressModifierInterface $orderAddressModifier;
+
+    private OrderPromoCodeAssignerInterface $orderPromoCodeAssigner;
+
+    public function __construct(
+        OrderRepositoryInterface $orderRepository,
+        OrderAddressModifierInterface $orderAddressModifier,
+        OrderPromoCodeAssignerInterface $orderPromoCodeAssigner
+    ) {
+        $this->orderRepository = $orderRepository;
+        $this->orderAddressModifier = $orderAddressModifier;
+        $this->orderPromoCodeAssigner = $orderPromoCodeAssigner;
+    }
+
+    public function __invoke(UpdateCart $updateCart): OrderInterface
+    {
+        $tokenValue = $updateCart->getOrderTokenValue();
+
+        /** @var OrderInterface|null $order */
+        $order = $this->orderRepository->findOneBy(['tokenValue' => $tokenValue]);
+        Assert::notNull($order, sprintf('Order with %s token has not been found.', $tokenValue));
+
+        if ($updateCart->getBillingAddress()) {
```

Comment:
Why did you decide for this condition? Shouldn't we process it for both billing and shipping address?

Reply:
@lchrusciel since before this change there was a special endpoint for addressing, it was always applying address transition in state machine. But now we use this endpiont for other changes too, so my logic was to check if address is sent in the request (assuming billing address is required) in order to know I need to trigger addressing transition and make changes to address. Not sure if I got the addressing logic correctly :) when the addressing transition should be triggered?

Reply:
makes sense 👍